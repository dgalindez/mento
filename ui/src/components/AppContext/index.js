import { createContext, useEffect } from 'react';

import useGeolocation from '../../hooks/useGeolocation';
import useLocale from '../../hooks/useLocale';
import useWeatherApi from '../../hooks/useWeatherApi';

const AppContext = createContext({
  geolocation: {},
  locale: {},
  weather: {},
});

export const ContextComponent = ({ children }) => {
  const geolocation = useGeolocation();
  const locale = useLocale();
  const weather = useWeatherApi();

  useEffect(() => {
    if (!geolocation?.failed) {
      weather?.getLocationData(geolocation.lat, geolocation.lon);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [geolocation.failed, geolocation.lat, geolocation.lon]);

  return (
    <AppContext.Provider value={{ geolocation, locale, weather }}>
      {children}
    </AppContext.Provider>
  );
};

export default AppContext;
