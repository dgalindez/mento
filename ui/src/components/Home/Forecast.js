import { useContext } from 'react';
import { useTranslation } from 'react-i18next';

import { UNIT_SYMBOLS } from '../../utils/constants';
import AppContext from '../AppContext';

const Forecast = () => {
  const { t } = useTranslation();

  const {
    locale: {
      units,
    },
    weather: {
      daily,
    },
  } = useContext(AppContext);

  return (
    <div className="column">
      <table className="table">
        <thead>
          <tr>
            <th>{t('forecast.date')}</th>
            <th>{t('forecast.max')} / {t('forecast.min')}</th>
          </tr>
        </thead>
        <tbody>
          {daily.map(({ date, max, min }, index) => (
          <tr key={`detail-day-${index}`}>
            <td>{new Date(date).toLocaleDateString()}</td>
            <td>{max} {UNIT_SYMBOLS[units]} / {min} {UNIT_SYMBOLS[units]}</td>
          </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Forecast;
