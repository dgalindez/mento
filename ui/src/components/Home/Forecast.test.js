import { render, screen } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';

import i18n from '../../utils/i18nTestHelper';
import AppContext from '../AppContext';

import Forecast from './Forecast';

describe('Forecast', () => {
  let contextData = {};

  beforeEach(() => {
    contextData = {
      locale: {
        units: 'metric',
      },
      weather: {
        daily: [
          {
            max: 22.35,
            min: 14.75,
            date: 'Tue, 05 Jul 2022 06:00:00 GMT'
          },
          {
            max: 22.04,
            min: 13.8,
            date: 'Wed, 06 Jul 2022 00:00:00 GMT'
          },
          {
            max: 21.08,
            min: 12.85,
            date: 'Thu, 07 Jul 2022 00:00:00 GMT'
          },
          {
            max: 24.47,
            min: 12.91,
            date: 'Fri, 08 Jul 2022 00:00:00 GMT'
          },
          {
            max: 25.43,
            min: 13.72,
            date: 'Sat, 09 Jul 2022 00:00:00 GMT'
          },
          {
            max: 21.71,
            min: 17.79,
            date: 'Sun, 10 Jul 2022 00:00:00 GMT'
          },
        ],
      },
    }
  });

  it('renders correctly', () => {
    render(
      <I18nextProvider i18n={i18n}>
        <AppContext.Provider value={contextData}>
          <Forecast />
        </AppContext.Provider>
      </I18nextProvider>
    );
    expect(screen.getByRole('table')).toMatchSnapshot();
  })
});
