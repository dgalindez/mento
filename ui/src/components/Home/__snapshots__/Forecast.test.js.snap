// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast renders correctly 1`] = `
<table
  class="table m-4 mx-6"
>
  <thead>
    <tr>
      <th>
        forecast.date
      </th>
      <th>
        forecast.max
         / 
        forecast.min
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        7/5/2022
      </td>
      <td>
        22.35
         
        ℃
         / 
        14.75
         
        ℃
      </td>
    </tr>
    <tr>
      <td>
        7/5/2022
      </td>
      <td>
        22.04
         
        ℃
         / 
        13.8
         
        ℃
      </td>
    </tr>
    <tr>
      <td>
        7/6/2022
      </td>
      <td>
        21.08
         
        ℃
         / 
        12.85
         
        ℃
      </td>
    </tr>
    <tr>
      <td>
        7/7/2022
      </td>
      <td>
        24.47
         
        ℃
         / 
        12.91
         
        ℃
      </td>
    </tr>
    <tr>
      <td>
        7/8/2022
      </td>
      <td>
        25.43
         
        ℃
         / 
        13.72
         
        ℃
      </td>
    </tr>
    <tr>
      <td>
        7/9/2022
      </td>
      <td>
        21.71
         
        ℃
         / 
        17.79
         
        ℃
      </td>
    </tr>
  </tbody>
</table>
`;
